multiqc/modules/nonpareil/nonpareil.py

```diff
@@ -0,0 +1,276 @@
+        self.nonpareil_general_stats_table()
+
+        # Add plots
+        self.nonpareil_redundancy_plot()
+
+    def parse_nonpareil_json(self, f):
+        """Parse the JSON output from nonpareil"""
+        import json
+
+        try:
+            json_raw = json.load(f["f"])
+        except json.JSONDecodeError as e:
+            log.warning(f"Could not parse nonpareil JSON: '{f['fn']}': {e}, skipping file")
+            return None, {}
+
+        # x.obs Rarefied sequencing effort.
+        # x.adj Adjusted rarefied sequencing effort.
+        # y.red Rarefied redundancy (observed).
+        # y.cov Rarefied coverage (corrected).
+        # y.sd Standard deviation of rarefied coverage.
+        # y.p25 Percentile 25 (1st quartile) of rarefied coverage.
+        # y.p50 Percentile 50 (median) of rarefied coverage.
+        # y.p75 Percentile 75 (3rd quartile) of rarefied coverage
+
+        import numpy as np
+
+        for s_name, data in json_raw.items():
+            # Conbert base pairs to gigabase pairs
+            data["LR"] = data["LR"] / 1000 / 1000
+            data["LRstar"] = data["LRstar"] / 1000 / 1000
+            data["x.adj"] = [(1e-6 if x == 0 else x) / 1000 / 1000 for x in data["x.adj"]]
+            data["x.model"] = [x / 1000 / 1000 for x in data["x.model"]]
+            data["observed"] = {x: y for x, y in zip(data["x.adj"], data["y.cov"])}
+            data["model"] = {x: y for x, y in zip(data["x.model"], data["y.model"])}
+            # Calculate dispersion
+            # from https://github.com/lmrodriguezr/nonpareil/blob/162f1697ab1a21128e1857dd87fa93011e30c1ba/utils/Nonpareil/R/Nonpareil.R#L306-L318
+            disp_add = False
+            if self.disp_type == "sd":
+                disp_add = data["y.sd"]
+            elif self.disp_type == "ci95":
+                disp_add = list(np.array(data["y.sd"]) * 1.9)
+            elif self.disp_type == "ci90":
+                disp_add = list(np.array(data["y.sd"]) * 1.64)
+            elif self.disp_type == "ci50":
+                disp_add = list(np.array(data["y.sd"]) * 0.67)
+            elif self.disp_type == "iq":
+                data[f"{self.disp_type}_upper"] = {x: y for x, y in zip(data["x.adj"], data["y.p75"])}
+                data[f"{self.disp_type}_lower"] = {x: y for x, y in zip(data["x.adj"], data["y.p25"])}
+
+            if disp_add:
+                data[f"{self.disp_type}_upper"] = {
+                    x: y for x, y in zip(data["x.adj"], list(np.array(data["y.cov"]) + np.array(disp_add)))
+                }
+                data[f"{self.disp_type}_lower"] = {
+                    x: y for x, y in zip(data["x.adj"], list(np.array(data["y.cov"]) - np.array(disp_add)))
+                }
+
+        return json_raw
+
+    def nonpareil_general_stats_table(self):
+        """Take the parsed stats from the nonpareil report and add it to the
+        General Statistics table at the top of the report"""
+
+        headers = {
+            "L": {
```

Comment:
I'd try to rename the keys as well to something a bit longer and readable, like `read_length`. They end up in various places and I'd prefer to still keep them human-readable of possible

Reply:
But don't these names need to match the `self.data_by_sample` keys?

Reply:
They do, but you can put them into `self.data_by_sample` with any keys you want :)

Reply:
But those names are the ones that `nonpareil` uses and I read all at once from the `json`. Since these names are not visible to the user, don't you think it is better to keep them the same as `nonpareil`?
This way it is easier to know where the number are coming from...

Reply:
They are used in `multiqc_data.json` which can be used downstream, and they are visible in the detailed view: 
<img width="745" alt="Screenshot 2023-12-14 at 16 46 54" src="https://github.com/ewels/MultiQC/assets/1575412/8dfb43ab-f216-4505-96ee-3a47da71e3cb">
But I see your point - that's actually fair. I haven't though that it's really helpful to track numbers down to the source json. Let's keep them then!

Reply:
@vladsavelyev are they scoped in the JSON? I feel like this is a name collision waiting to happen if not. Shouldn't be too difficult to add a prefix after parsing the JSON, that would feel safer..

Reply:
Prefix added.